models/shopcart.py

```diff
@@ -53,13 +53,19 @@ def deserialize(self,data):
                 if type( data['products'] ) == dict:
                     # ** JSON.DUMP MAKES KEYS TO STR
                     prods = { int(p):int(q) for (p,q) in data['products'].items() }
+                    for pid,quant in prods.items():
+                        if quant == 0:
+                            self.delete_product(pid)
+                            prods.pop(pid,None)
                 else:
```

Comment:
I'm getting the `Updating a product to 0` test failing at this part:
`for pid,quant in prods.items():` 
with the error:
`RuntimeError: dictionary changed size during iteration`

Reply:
I'll check it! thanks

Reply:
I don't know how i missed that. Thanks @radhikamattoo

Reply:
I don't know how i missed that. Thanks @radhikamattoo